### PR TITLE
RD-6005: Adjustments for using multipart form data

### PR DESCRIPTION
**Pull request by Vorbert-Kruk**
## Description

This PR is a continuation of #2171.
Within this PR [comments from the #2171](https://github.com/cloudify-cosmo/cloudify-stage/pull/2171#pullrequestreview-1127205697) should be addressed 💪  

## Screenshots / Videos

N/A

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

According to the recommendation, they're being executed 😁 

[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3632/)

## Documentation

N/A

### CODE REVIEW

**Comment by Vorbert-Kruk** [comment]
@qooban 
Most of the comments from #2171 were applied, but one may still need a discussion 😁
Because of that, I'm pasting your question, to continue discussion under this PR

[Link to the question](https://github.com/cloudify-cosmo/cloudify-stage/pull/2171#discussion_r984854946)
Question:
> Taking into account note from L247-249 browser will adjust content-type not only for multipart data, but also when we are sending file as payload? (so it'll still be application/octet-stream even though not explicitly provided) 


I've tested it on uploading snapshots (it's being done via zip file) and the content-type was `application/x-zip-compressed`, so it's a little bit different

WDYT, we should be good to leave it as it is, or should I bring back 
```typescript
xhr.setRequestHeader('content-type', 'application/octet-stream');
```
and remove it upon sonding a `multipart/form-data`? 🤔

File: widgets/common/src/blueprints/BlueprintActions.ts [hunk 1]
```diff
@@ -135,6 +144,11 @@ export const CompletedBlueprintStates = {
     Invalid: 'invalid'
 };
 
+const UploadBlueprintFormFieldNames = {
```
Comment (Vorbert-Kruk) [inline widgets/common/src/blueprints/BlueprintActions.ts:147]: ### Note
I couldn't think of a better name for this const 😅 
If you'd have any naming suggestions feel free to comment them 💪

**Review by qooban — COMMENTED** [review]
Looks good overall, added just 2 comments.

I think it's worth to trigger UI system tests to be sure all is fine.

> WDYT, we should be good to leave it as it is, or should I bring back
> ```
> xhr.setRequestHeader('content-type', 'application/octet-stream');
> ```
> and remove it upon sonding a multipart/form-data? thinking

If it works fine for different cases (uploading blueprints, snapshots, widgets), then I think we can leave it as is.

File: backend/package-lock.json [hunk 2]
```diff
@@ -23381,7 +23381,7 @@
             "resolved": "https://registry.npmjs.org/axios/-/axios-0.27.2.tgz",
             "integrity": "sha512-t+yRIyySRTp/wua5xEr+z1q60QmLq8ABsS5O9Me1AsE5dfKqgnCFzwiCZZ/cGNd1lq4/7akDWMxdhVlucjmnOQ==",
             "requires": {
-                "follow-redirects": "^1.14.9",
+                "follow-redirects": "^1.15.2",
```
[the remark this answers is not in the document]
Comment (Vorbert-Kruk) [reply backend/package-lock.json:23384]: Oh, you're right
Just to be safe, I've reverted all of the `backend/package-lock.json` modifications delivered within #2171 💪

File: widgets/common/src/blueprints/BlueprintActions.ts (hunk 1 above)
Comment (Vorbert-Kruk) [reply widgets/common/src/blueprints/BlueprintActions.ts]: Both of them are looking much better!
I think that I'll go with `UploadBlueprintFormDataProperties`

Thanks for the suggestions 🎉

**Comment by Vorbert-Kruk** [comment]
> I think it's worth to trigger UI system tests to be sure all is fine. 

Sure, sounds reasonable
[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3631/)

>> WDYT, we should be good to leave it as it is, or should I bring back
>>xhr.setRequestHeader('content-type', 'application/octet-stream');
and remove it upon sonding a multipart/form-data? thinking
>
> If it works fine for different cases (uploading blueprints, snapshots, widgets), then I think we can leave it as is.

I didn't test it with widgets, as I don't have a required, built widget package
But I've tested snapshots and they seem to be uploading just fine 🎉 
So I think that we should be good to go with that change 🚀

**Comment by Vorbert-Kruk** [comment]
[System tests](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3632/) has been executed
The same test scenarios are failing as in [nightly](https://jenkins.cloudify.co/job/Stage-UI-System-Test/3630/), so I consider the [current execution](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/3632/) a success

**Review by qooban — APPROVED** [review]

**Pull request merged**